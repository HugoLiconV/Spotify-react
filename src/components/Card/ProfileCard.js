import React from 'react';
import Button from '../Button/Button';
import Card from './Card';
import CardAvatar from './CardAvatar';
import CardBody from './CardBody';
import { withStyles } from '@material-ui/core';
import PropTypes from 'prop-types';
import noImageFound from '../../assets/img/no_image_found.png';

const styles = {
  details: {
    display: 'flex',
    flexWrap: 'wrap',
    justifyContent: 'center',
    '& *': {
      margin: '10px'
    }
  }
};

const currentUser = {
  country: 'MX',
  display_name: 'Hugo Licon',
  external_urls: {
    spotify: 'https://open.spotify.com/user/hugofernandoliconvalenzuela'
  },
  followers: {
    href: null,
    total: 12
  },
  images: [
    {
      height: null,
      url:
        'https://scontent.xx.fbcdn.net/v/t1.0-1/p200x200/11880385_912284995474857_7569056323770464976_n.jpg?_nc_cat=109&_nc_ht=scontent.xx&oh=8dea37fe4bd092920aa0a9277f85d1f0&oe=5CD0F3C5',
      width: null
    }
  ],
  product: 'premium',
  id: 'hugofernandoliconvalenzuela'
};

const getProfilePicture = currentUser => {
  if (currentUser.images.length === 0) return noImageFound;
  return currentUser.images[0].url;
};
export const ProfileCard = props => {
  const { currentUser, classes } = props;
  return (
    <Card profile>
      <CardAvatar profile>
        <a href={currentUser.external_urls.spotify}>
          <img
            style={{ backgroundColor: 'white' }}
            src={getProfilePicture(currentUser)}
            alt={currentUser.display_name}
          />
        </a>
      </CardAvatar>
      <CardBody profile>
        <div className={classes.details}>
          <h6 className={classes.cardCategory} data-product>
            Product: {currentUser.product}
          </h6>
          <h6 className={classes.cardCategory} data-country>
            Country: {currentUser.country}
          </h6>
          <h6 className={classes.cardCategory} data-followers>
            Followers: {currentUser.followers.total}
          </h6>
        </div>
        <h4 className={classes.cardTitle} data-name>
          {currentUser.display_name}
        </h4>
        <Button color="primary" round href={currentUser.external_urls.spotify}>
          Account
        </Button>
      </CardBody>
    </Card>
  );
};

ProfileCard.propTypes = {
  classes: PropTypes.object.isRequired,
  currentUser: PropTypes.object.isRequired
};

ProfileCard.defaultProps = {
  currentUser
};
export default withStyles(styles)(ProfileCard);
