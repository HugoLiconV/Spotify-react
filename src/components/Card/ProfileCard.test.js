import React from 'react';
import { ProfileCard } from './ProfileCard';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('<ProfileCard /> component', () => {
  it('should render default Props', () => {
    const wrapper = shallow(<ProfileCard classes={{}} />);
    const product = wrapper.find('[data-product]');
    const country = wrapper.find('[data-country]');
    const followers = wrapper.find('[data-followers]');
    const name = wrapper.find('[data-name]');

    expect(product.text()).toBe('Product: premium');
    expect(country.text()).toBe('Country: MX');
    expect(followers.text()).toBe('Followers: 12');
    expect(name.text()).toBe('Hugo Licon');
  });

  it('should render props', () => {
    const currentUser = {
      country: 'MX',
      display_name: 'Hugo Licon',
      external_urls: {
        spotify: 'https://open.spotify.com/user/hugofernandoliconvalenzuela'
      },
      followers: {
        href: null,
        total: 12
      },
      images: [
        {
          height: null,
          url:
            'https://scontent.xx.fbcdn.net/v/t1.0-1/p200x200/11880385_912284995474857_7569056323770464976_n.jpg?_nc_cat=109&_nc_ht=scontent.xx&oh=8dea37fe4bd092920aa0a9277f85d1f0&oe=5CD0F3C5',
          width: null
        }
      ],
      product: 'premium',
      id: 'hugofernandoliconvalenzuela'
    };
    const wrapper = shallow(
      <ProfileCard classes={{}} currentUser={currentUser} />
    );
    const product = wrapper.find('[data-product]');
    const country = wrapper.find('[data-country]');
    const followers = wrapper.find('[data-followers]');
    const name = wrapper.find('[data-name]');

    expect(product.text()).toBe(`Product: ${currentUser.product}`);
    expect(country.text()).toBe(`Country: ${currentUser.country}`);
    expect(followers.text()).toBe(`Followers: ${currentUser.followers.total}`);
    expect(name.text()).toBe(currentUser.display_name);
  });

  it('should match the snapshot', () => {
    const tree = shallow(<ProfileCard classes={{}} />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
