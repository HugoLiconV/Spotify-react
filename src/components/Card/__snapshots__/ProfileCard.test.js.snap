// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfileCard /> component should match the snapshot 1`] = `
<WithStyles(Card)
  profile={true}
>
  <WithStyles(CardAvatar)
    profile={true}
  >
    <a
      href="https://open.spotify.com/user/hugofernandoliconvalenzuela"
    >
      <img
        alt="Hugo Licon"
        src="https://scontent.xx.fbcdn.net/v/t1.0-1/p200x200/11880385_912284995474857_7569056323770464976_n.jpg?_nc_cat=109&_nc_ht=scontent.xx&oh=8dea37fe4bd092920aa0a9277f85d1f0&oe=5CD0F3C5"
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      />
    </a>
  </WithStyles(CardAvatar)>
  <WithStyles(CardBody)
    profile={true}
  >
    <div>
      <h6
        data-product={true}
      >
        Product: 
        premium
      </h6>
      <h6
        data-country={true}
      >
        Country: 
        MX
      </h6>
      <h6
        data-followers={true}
      >
        Followers: 
        12
      </h6>
    </div>
    <h4
      data-name={true}
    >
      Hugo Licon
    </h4>
    <WithStyles(RegularButton)
      color="primary"
      href="https://open.spotify.com/user/hugofernandoliconvalenzuela"
      round={true}
    >
      Account
    </WithStyles(RegularButton)>
  </WithStyles(CardBody)>
</WithStyles(Card)>
`;
